en/concept/architecture.md

```diff
@@ -1,36 +1,121 @@
 # Architectural Overview
 
-PX4 consists of two main layers: The [PX4 flight stack](../concept/flight_stack.md), an autopilot software solution and the [PX4 middleware](../concept/middleware.md), a general robotics middleware which can support any type of autonomous robot.
+PX4 consists of two main layers: the [flight stack](#flight-stack) is an autopilot software solution,
+and the [middleware](#middleware) is a general robotics layer that can support any type of autonomous robot.
 
-All [airframes](../airframes/architecture.md), and in fact all robotic systems including boats, share a single codebase. The complete system design is [reactive](http://www.reactivemanifesto.org), which means that:
+All PX4 [airframes](../airframes/README.md) share a single codebase (this includes other robotic systems like boats, rovers, submarines etc.). The complete system design is [reactive](http://www.reactivemanifesto.org), which means that:
 
-  * All functionality is divided into exchangeable components
-  * Communication is done by asynchronous message passing 
-  * The system can deal with varying workload
+- All functionality is divided into exchangeable components
+- Communication is done by asynchronous message passing 
+- The system can deal with varying workload
 
 In addition to these runtime considerations, its modularity maximizes [reusability](https://en.wikipedia.org/wiki/Reusability).
 
+
 ## High Level Software Architecture
 
-Each of the blocks below is a separate module, which is self-contained in terms of code, dependencies and even at runtime. Each arrow is a connection through publish/subscribe calls through [uORB](../middleware/uorb.md).
+The diagram below provides an overview of the building blocks of PX4. 
+The top part of the diagram contains middleware blocks, while the lower
+section shows the components of the flight stack.
+
+![PX4 Architecture](../../assets/diagrams/PX4_Architecture.svg)
+
+<!-- This diagram can be updated from 
+[here](https://drive.google.com/file/d/0B1TDW9ajamYkaGx3R0xGb1NaeU0/view?usp=sharing) 
+and opened with draw.io Diagrams. -->
+
+The source code is split into self-contained modules (shown in `monospace` in the
```

Comment:
module is a much overloaded software term. If you mean "executable" or program it is worth adding that. E.g. modules/programs (shown in ...).